src/data-access/user.dao.ts

```diff
@@ -27,18 +53,111 @@ export class UserDAO {
     }
 
     getUserById(id: string): Promise<User | null> {
-        return this._userModel.findOne({ where: { id } });
+        return this._usersModel.findOne({ ...includeOption, where: { id } });
     }
 
-    saveUser(user: User): Promise<User> {
-        return this._userModel.create(user);
+    async saveUser(reqUser: UserCreateRequest): Promise<User> {
+        const { login, password, age } = reqUser;
+        const groupIds: string[] = reqUser.groups;
+
+        const user: User = {
+            login,
+            password,
+            age
+        };
+
+        try {
+            const result = await db.transaction(async () => {
+                const savedUser: User = await this._usersModel.create(user);
+
+                for (let groupId of groupIds) {
+                    const foundGroup: Group | null = await this._groupDao.getGroupById(
+                        groupId
+                    );
+
+                    if (!foundGroup) {
+                        throw 'No such group';
+                    }
+
+                    await this._usersGroupsDAO.saveUserGroup(
+                        <string>savedUser.id,
+                        foundGroup.id
+                    );
+                }
```

Comment:
You don't need to manage it manually, just `await savedUser.addGroups(groupIds)`